packages/network/test/integration/proxy-connections.test.ts

```diff
@@ -68,93 +73,82 @@ describe('Proxy connection tests', () => {
             tracker?.stop(),
             onewayNode?.stop(),
             contactNode?.stop(),
-            contactNode2?.stop(),
+            contactNode2?.stop()
         ])
     })
 
     it('publisher node can form proxy connections', async () => {
-        await onewayNode.openProxyConnection(defaultStreamPartId, 'contact-node', ProxyDirection.PUBLISH, 'publisher')
-        await onewayNode.openProxyConnection(defaultStreamPartId, 'contact-node-2', ProxyDirection.PUBLISH, 'publisher')
+        await onewayNode.setProxies(streamPartId, ['contact-node', 'contact-node-2'], ProxyDirection.PUBLISH, async () => 'publisher')
         // @ts-expect-error private
-        expect(onewayNode.streamPartManager.getOutboundNodesForStreamPart(defaultStreamPartId)).toContainValues(['contact-node', 'contact-node-2'])
+        expect(onewayNode.streamPartManager.getOutboundNodesForStreamPart(streamPartId)).toContainValues(['contact-node', 'contact-node-2'])
```

Comment:
The assertions are quite long. We could maybe use helper function to make these more readable?

E.g.
```
const state = getState(onewayNode)
expect(state.outboundNodes).toContainValues(['contact-node', 'contact-node-2'])
expect(state.neighbours).toBeArrayOfSize(0)
```

The `getState` could be something like this:
```
const getState = (node: NetworkNode, ) => {
    // @ts-expect-error private
    const streamPartManager = node.streamPartManager
    if (streamPartManager.isSetUp(streamPartId)) {
        return {
            isSetUp: true,
            inboundNodes: streamPartManager.getInboundNodesForStreamPart(streamPartId),
            outboundNodes: streamPartManager.getOutboundNodesForStreamPart(streamPartId),
            neighbours: streamPartManager.getNeighborsForStreamPart(streamPartId),
            hasInOnlyConnection: (otherNodeId: NodeId) => streamPartManager.hasInOnlyConnection(streamPartId, otherNodeId),
            hasOutOnlyConnection: (otherNodeId: NodeId) => streamPartManager.hasOutOnlyConnection(streamPartId, otherNodeId),
        }
    } else {
        return {
            isSetUp: false
        }
    }
}
```